src/biome/text/modules/heads/token_classification.py

```diff
@@ -142,28 +140,31 @@ def featurize(
 
             They are used with the `spacy.gold.biluo_tags_from_offsets` method.
         """
+        if isinstance(text, str):
+            doc = self.backbone.tokenizer.nlp(text)
+            tokens = [token.text for token in doc]
```

Comment:
This basically substitutes the tokenization step otherwise done in `self.backbone.featurizer(text, to_field="text", tokenize=True, aggregate=True)`, right? However, in `self.backbone.featurizer` we do a lot of checks on how to exactly tokenize the data, not sure how this influences all the different possibilities ... did you check this?

Reply:
For token classification, if we must tokenize the original text, we must control the whole tokenization process, keeping the original text accesible once the tokenization is done.

Our tokenizer incude pre and post tasks for text and token cleaning that we must by-pass in token classification. We continue using tokenizer featurization, just skipping tokenization phase, like with pre-tokenized text.